src/nnet3/convolution-cudnn.cc

```diff
@@ -232,35 +231,30 @@ void ConvolutionComputation::InitCudnn() {
   // These two member functions depend only on input_desc_,
   // conv_desc_, and params_desc_, so they are safe to call now.
   int out_dims[4] = {c.num_images, c.num_channels_out, c.output_image_width,
-		     c.output_image_height};
+                     c.output_image_height};
   int out_stride[4] = {c.num_channels_out * c.output_image_width * c.output_image_height,
-		       c.output_image_width * c.output_image_height,
-		       c.output_image_height, 1};
+                       c.output_image_width * c.output_image_height,
+                       c.output_image_height, 1};
   CUDNN_SAFE_CALL(
     cudnnSetTensorNdDescriptor(output_desc_, CUDNN_DATA_BASEFLOAT, 4, out_dims,
-			       out_stride));
+                               out_stride));
 
   // We pad the bias with leading dims of 1, since CUDNN's tensors appear to
   // need a dimension of at least 3.
-  int bias_dims[3] = {1, 1, c.num_channels_out};
-  int bias_stride[3] = {c.num_channels_out, c.num_channels_out, 1};
+  int bias_dims[4] = {1, c.num_channels_out, 1, 1};
```

Comment:
Can you explain - preferably in a comment - why this specific pattern of dims is required? It's mysterious to me.

Reply:
Done. I can make the bias optional at some point later.